src/test/java/com/marklogic/mule/extension/WriteDocumentTest.java

```diff
@@ -179,4 +179,32 @@ public void writeDocumentWithUuid() {
                 .verify();
         }
     }
+
+    @Test
+    public void writeDocumentWithTemporalCollection() {
```

Comment:
This test is a little confusing because it's not just a simple "read a doc, write is a temporal doc, verify the temporal doc". While the "delete-docs" operation could be included in a separate test, I don't think we need it because we're effectively testing MarkLogic, not our connector - i.e. we're verifying that MarkLogic doesn't let you do an ordinary delete on a temporal document. 

The main thing to test on a temporal document is the set of collections. When a document is inserted into a temporal collection, MarkLogic will also add it to a collection named "latest" and a collection named after the URI of the document as well. Those assertions need to be added. I would remove the assertion on quality and just not set it at all because it's only adding noise to the test. 

Also, for any test that we expect to throw an exception, `assertThrows` should be used. I was curious what would happen if I removed the "delete-docs" from this flow, and the test still passed. It would have failed if `assertThrows` were used instead.